### PR TITLE
V6 migration

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
We need a v6 migration guide!

### How was it fixed?
Added one! 

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: docs/v6_migration.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Exceptions inherit from a base class
+------------------------------------
+
+In v5, some Web3.py exceptions inherited from ``AttributeError``, namely:
+
+- ``NoABIFunctionsFound``
+- ``NoABIFound``
+- ``NoABIEventsFound``
+
+Others inherited from ValueError, namely:
```
Comment (fselmo) [inline docs/v6_migration.rst:47]: nit, for consistency with the above example

```suggestion
Others inherited from ``ValueError``, namely:
```

File: docs/v6_migration.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+- ``InvalidEventABI``
+
+Now Web3.py exceptions inherit from the same base ``Web3Exception``.
+
+As such, any code that was expecting a ``ValueError`` or an ``AttributeError`` from
+Web3.py must update to expecting one of the exception listed above, or
+``Web3Exception``.
+
+Similarly, exceptions raised in the EthPM and ENS modules inherit from the base
+``PyEthPMException`` and ``ENSException``, respectively.
```
Comment (fselmo) [inline docs/v6_migration.rst:61]:
```suggestion
``EthPMException`` and ``ENSException``, respectively.
```

File: docs/v6_migration.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+- ``InvalidAddress``
+- ``NameNotFound``
+- ``LogTopicError``
+- ``InvalidEventABI``
+
+Now Web3.py exceptions inherit from the same base ``Web3Exception``.
+
+As such, any code that was expecting a ``ValueError`` or an ``AttributeError`` from
+Web3.py must update to expecting one of the exception listed above, or
```
Comment (fselmo) [inline docs/v6_migration.rst:57]:
```suggestion
Web3.py must update to expecting one of the exceptions listed above, or
```

**Review by fselmo — COMMENTED** [review]
I just found some minor tweaks. Are we trying to leave this branch open, add to it, and only merge when it's finalized?

**Comment by kclowes** [comment]
> Are we trying to leave this branch open, add to it, and only merge when it's finalized?

I was thinking we'd get it in now-ish and then add to it as breaking changes come up, but open to either if you have a strong opinion.

**Comment by fselmo** [comment]
> I was thinking we'd get it in now-ish and then add to it as breaking changes come up, but open to either if you have a strong opinion.

No strong opinion here, just seeing that it was still a draft but had a review request haha. I think the fact we're in beta is good enough to merge this and keep adding to it as the PRs come in. I think I'd even prefer merging this in since that gives users more time to make the changes that are already on here.

**Comment by kclowes** [comment]
> I think I'd even prefer merging this in since that gives users more time to make the changes that are already on here.

Same! Updated! Lmk if there are changes that should be made to clarify the ValidationError section. That one was sort of tricky to describe.

**Comment by fselmo** [comment]
> Lmk if there are changes that should be made to clarify the ValidationError section

I think this looks good for a migration guide description. We can go a bit more in depth with the actual documentation on it once that comes along.

**Review by pacrob — APPROVED** [review]
lgtm

**Pull request merged**